src/NServiceBus.Gateway/IGatewayDeduplicationPersistence.cs

```diff
@@ -0,0 +1,30 @@
+﻿namespace NServiceBus
+{
+    using System.Threading.Tasks;
+    using System.Transactions;
+    using Extensibility;
+
+    /// <summary>
+    /// Defines the api to provide storage for the gateway deduplication mechanism.
+    /// </summary>
+    public interface IGatewayDeduplicationStorage
+    {
+        /// <summary>
+        /// Returns whether the storage can enlist in a distributed transaction via ambient <see cref="TransactionScope"/> to support exactly-once dispatching.
+        /// </summary>
+        bool SupportsDistributedTransactions { get; }
+
+        /// <summary>
+        /// Returns if the message is a duplicate.
```

Comment:
```suggestion
        /// Returns if the message is a duplicate. Should INSERT in TX context to optimize for happy path.
```

Reply:
That's a very implementation specifc comment which shouldn't be here imo.